jenkins/Jenkinsfile

```
@@ -36,41 +45,79 @@ pipeline {
     timestamps()
   }
   stages {
-    stage('Build') {
-      steps {
-        container('node'){
-          dir("${env.WORKSPACE}/${env.PROJECT}") {
-            echo "Install dependencies"
-            sh 'npm ci --prefer-offline'
+    stage('CI & Audit'){
+      parallel{
+        stage('Audit'){
+          when {
+            branch 'master'
+          }
+          steps {
+            container('node'){
+              dir("${env.WORKSPACE}/${env.PROJECT}") {
+                echo 'Run npm production dependencies security audit'
+                sh 'npm run audit'
+              }
+            }
+          }
+        }
+        stage('Clean Install') {
+          steps {
+            container('node'){
+              dir("${env.WORKSPACE}/${env.PROJECT}") {
+                restoreArchiveFolder('~/.npm', "${env.WORKSPACE}/${env.PROJECT}/package-lock.json", 'components-mb', 'npm_dir', '/mnt/data')
+                echo "Install dependencies"
+                sh 'npm ci --prefer-offline'
+                archiveFolder('~/.npm', "${env.WORKSPACE}/${env.PROJECT}/package-lock.json", 'components-mb', 'npm_dir', '/mnt/data')
+              }
+            }
+          }
+        }
+      }
+    }
+    stage('Run Build & Tests'){
```

Maybe to keep the same naming convention for all stages we shouldn't use "Run"?
```suggestion
    stage('Build & Tests'){
```